src/vsr/replica.zig

```diff
@@ -5287,7 +5356,12 @@ pub fn ReplicaType(
                             },
                         );
                     }
+
                     self.replace_headers(message_headers);
+                    op_canonical = std.math.max(
+                        op_canonical,
+                        do_view_change_headers_gap_min(message) orelse message_headers[0].op,
```

Comment:
Could we have `do_view_change_headers_gap_min()` handle the `orelse` branch internally?

Alternatively, can we do more here to assert that the first header in `message_headers` is in fact the highest op?